app-games/src/main/java/com/aptoide/android/aptoidegames/gamegenie/data/database/model/Utils.kt

```diff
@@ -0,0 +1,37 @@
+package com.aptoide.android.aptoidegames.gamegenie.data.database.model
+
+import com.aptoide.android.aptoidegames.gamegenie.domain.ChatInteraction
+import com.aptoide.android.aptoidegames.gamegenie.domain.GameContext
+import com.aptoide.android.aptoidegames.gamegenie.domain.GameGenieChat
+import com.google.common.reflect.TypeToken
+import com.google.gson.Gson
+
+fun GameGenieHistoryEntity.toDomain(): GameGenieChat {
+  return GameGenieChat(
+    id = this.id,
+    conversation = this.conversation.map { it.toDomain() }
+  )
+}
+
+fun ChatInteractionEntity.toDomain(): ChatInteraction {
+  return ChatInteraction(
+    gpt = this.gpt,
+    user = this.user,
+    apps = Gson().fromJson(this.apps, object : TypeToken<List<GameContext>>() {}.type)
+  )
+}
+
+fun GameGenieChat.toEntity(): GameGenieHistoryEntity {
```

Comment:
This one is used only in the `ChatbotViewModel` so let's move it there for now.

In general we avoid creating any kind of "utils" or "tools" as much as possible preferring encapsulation instead.
Because these things are rarely generic enough to be reused in a different module or context,
But they result in pieces of code and logic scattered around making it harder to navigate the codebase.